list/list.go

```diff
@@ -455,12 +456,26 @@ func (m Model) MatchesForItem(index int) []int {
 	return m.filteredItems[index].matches
 }
 
-// Index returns the index of the currently selected item as it appears in the
-// entire slice of items.
+// Index returns the index of the currently selected item as it is stored in the
+// filtered list of items.
+// Using this value with SetItem() might be incorrect, consider using
+// GlobalIndex() instead.
```

Comment:
@g026r done in [11d4e11](https://github.com/charmbracelet/bubbles/pull/574/commits/11d4e111405f1b7677766270f6de79311e43bf7c)